gsy_framework/forward_markets/aggregated_ssp.py

```diff
@@ -0,0 +1,164 @@
+
+    def _load_aggregated_profiles(self) -> None:
+        """Load aggregated SSP profiles from disk."""
+
+        if self.SSP_AGGREGATED_AGGREGATED_PROFILE_PATH:
+            with open(self.SSP_AGGREGATED_AGGREGATED_PROFILE_PATH, encoding="utf-8") as inf:
+                reader = csv.DictReader(inf)
+                for row in reader:
+                    self._SSP_AGGREGATED_PROFILE[row["timeslot"]] = row["energy_kWh"]
+
+        if self.LEAP_YEAR_SSP_AGGREGATED_PROFILE_PATH:
+            with open(self.LEAP_YEAR_SSP_AGGREGATED_PROFILE_PATH, encoding="utf-8") as inf:
+                reader = csv.DictReader(inf)
+                for row in reader:
+                    self._LEAP_YEAR_SSP_AGGREGATED_PROFILE[row["timeslot"]] = row["energy_kWh"]
+
+    @abstractmethod
+    def _get_timeslots(self, start_time: DateTime, end_time: DateTime) -> Iterable:
+        """Return all timeslots with respect to start, end and resolution."""
+
+    @abstractmethod
+    def _get_timeslot_energy_kWh(self, timeslot: DateTime) -> float:
+        """Return the energy amount associated with the timeslot."""
+
+
+class QuarterHourAggregatedSSPProfile(AggregatedSSPProfileBase):
+    """Return 15-minute-aggregated profile of the SSP."""
+    def _get_timeslots(self, start_time: DateTime, end_time: DateTime) -> Iterable:
+        assert end_time - start_time >= duration(minutes=15), \
+            "Time period should be >= 15 minutes."
+        return period(
+            start=start_time.set(minute=(start_time.minute // 15) * 15),
+            end=end_time.set(minute=(end_time.minute // 15) * 15) - duration(minutes=15)
+        ).range("minutes", 15)
+
+    def _get_timeslot_energy_kWh(self, timeslot: DateTime) -> float:
+        return float(self._SSP_AGGREGATED_PROFILE[timeslot.month][timeslot.time()])
+
+    def _load_aggregated_profiles(self) -> None:
+        self._SSP_AGGREGATED_PROFILE = ProfileScaler(
+            StandardProfileParser().parse()).scale_by_peak(peak_kWh=1)
+
+
+class HourlyAggregatedSSPProfile(AggregatedSSPProfileBase):
+    """Return hourly-aggregated profile of the SSP."""
+    SSP_AGGREGATED_AGGREGATED_PROFILE_PATH = RESOURCES_PATH / "aggregated_ssp/hourly.csv"
+
+    def _get_timeslots(self, start_time: DateTime, end_time: DateTime) -> Iterable:
+        assert end_time - start_time >= duration(hours=1), "Time period should be >= 1 hour."
+        return period(
+            start=start_time.start_of("hour"),
+            end=end_time.start_of("hour") - duration(minutes=1)
+        ).range("hours", 1)
+
+    def _get_timeslot_energy_kWh(self, timeslot: DateTime) -> float:
+        return float(self._SSP_AGGREGATED_PROFILE[timeslot.format("M-H")])
+
+
+class WeeklyAggregatedSSPProfile(AggregatedSSPProfileBase):
+    """Return weekly-aggregated profile of the SSP."""
+
+    SSP_AGGREGATED_AGGREGATED_PROFILE_PATH = RESOURCES_PATH / "aggregated_ssp/daily.csv"
```

Comment:
Sorry, could you help me here, why we have daily ssp for weekly aggregation?

Reply:
This class uses the daily aggregation of the ssp to generate the weekly aggreation of it!
It wasn't possible to pre-calculate weekly aggregation of the ssp.